### DISCUSSION TITLE
Support for different sample rates

### DISCUSSION THREAD

**Discussion by danielschaeufele** [NVlabs/sionna#427] [General]
In a practical system the signal is usually upsampled before transmission (mainly to not violate the spectral emission mask). E.g., a signal with 100MHz bandwidth is usually transmitted with 122.88MHz sample rate.

Implementing this as a separate step is not terribly difficult (all that is needed is a larger FFT size and CP length with zero padding/removal), but it would be much more convenient to have support for this in `PUSCHTransmitter`/`PUSCHReceiver`. I could contribute an implementation if there is interest from the sionna team.

**Comment by jhoydis** [comment]
Hi @danielschaeufele,

That sounds probably like a nice feature. It would be desirable to have this available not only for the NR module but as a general block, maybe as an add-on for the OFDM modulation, demodulation blocks.
Alternatively, we could consider a resample block that would process time-domain signals.

**Reply by danielschaeufele** [reply]
I think the support is basically already there, it just needs to be enabled for `PUSCHTransmitter`/`PUSCHReceiver`. All that needs to change, would be to increase the FFT size and CP length for the `OFDMModulator`/`OFDMDemodulator` and then add the appropriate number of guard carriers in the `ResourceGrid`. 

I hope I can prepare some code next week.

**Reply by danielschaeufele** [reply]
I would propose implementing it like this: https://github.com/danielschaeufele/sionna/commit/9f2fbf51f0442cbf04533f52e8ce3274162f4af1
What do you think?

I still need to do some more testing, then I can create a PR if you think this is a good idea.
